freeglut/freeglut/src/mswin/fg_spaceball_mswin.c

```diff
@@ -56,3 +91,101 @@ void fgPlatformSpaceballSetWindow(SFG_Window *window)
 {
 	return;
 }
+
+int fgIsSpaceballWinEvent(HWND hwnd, WPARAM wParam, LPARAM lParam)
+{
+	return 0;
+}
+
+void fgSpaceballHandleWinEvent(HWND hwnd, WPARAM wParam, LPARAM lParam)
+{
+	#define LOGITECH_VENDOR_ID 0x46d
+	HRAWINPUT hRawInput = (HRAWINPUT)lParam;
+	UINT inputCode = (UINT)wParam;
+	UINT size;
+	BYTE *rawInputBuffer;
+	PRAWINPUT pRawInput;
+	UINT res;
+	RID_DEVICE_INFO sRidDeviceInfo;
+
+	if (!sball_initialized)
+	{
+		fgPlatformInitializeSpaceball();
+		if (!sball_initialized)
+		{
+			return;
+		}
+	}
+
+	res = GetRawInputData(hRawInput, RID_INPUT, NULL, &size, sizeof(RAWINPUTHEADER));
+	if (res == -1)
+		return;
+
+	rawInputBuffer = (BYTE*)malloc(size);
```

Comment:
Never cast the result of malloc, terrible habit, hides forgetting to include stdlib.h which will lead to crashes on systems where sizeof(int) != sizeof(void*)